### DISCUSSION TITLE
Enhancing docs when creating PRs?

### DISCUSSION THREAD

**Discussion by Jomik** [LazyVim/LazyVim#846] [General]
Hey @folke,
Love this project, and as you can see, want to contribute as much as possible.
One thing that still eludes me is how the docs generation work. I sort of understand how you are pulling out and showing the configurations, etc.
But _how_ are you creating the examples like the one for `neotest-plenary` ? https://www.lazyvim.org/plugins/extras/test.core
I cannot find mention of `neotest-plenary` anywhere in the source.

**Comment by folke** [comment]
The docs are all a bit black magic at the moment :)

There's a `build.lua` script in the LazyVim repo that generates everything. But it pretty mich only properly runs on my machine I guess.

It updates files in the webiste repo.

For the neotest one, check here https://github.com/LazyVim/lazyvim.github.io/blob/main/docs/plugins/extras/test.core.md?plain=1

Everything between `<!-- plugins:start -->` and `<!-- plugins:stop -->` is auto generated. 
The rest can be edited in place.

I'll see if I can move everything to the github.io repo, so that it's easy to generate the docs and create PR's for there as well.

**Reply by Jomik** [reply]
Oooh! I did find the `build.lua` script, just could not for the life of me see how the other stuff was added. It was manual stuff! 😄
Wonder if an easy way to have an option for that in the LazyVim repo, would be to accept a `plugin.name.md` file that it just adds above these `<!-- plugins:start -->` sections? Then it would probably be easier to maintain.
Though of course, would be best if docs was fully generated in pipeline - no idea if there is a good way to do that though 😄 
Let me know if this is something you want any help on 👍

**Reply by folke** [reply]
I just changed the whole docs building.

https://github.com/LazyVim/lazyvim.github.io now has a workflow that automatically generates the docs every hour and when pushed to main.

So if you want to change pages there, simply open a PR and after merging everything will be re-generated.
